src/pybind/mgr/dashboard/services/auth.py

```diff
@@ -43,7 +43,7 @@ def gen_token(cls, username):
             cls.init()
         ttl = mgr.get_module_option('jwt_token_ttl', cls.JWT_TOKEN_TTL)
         ttl = int(ttl)
-        now = int(time.mktime(time.gmtime()))
+        now = int(time.mktime(time.time()))
```

Comment:
The problem you are seeing is not caused by the `time.gmtime()` function. The problem is actually caused by the use of `time.mktime`, which according to https://docs.python.org/2/library/time.html#time.mktime expects a time tuple in local time.

We should use the `calendar.timegm` function to convert the time tuple into seconds UTC like this:

```suggestion
        now = int(calendar.timegm(time.gmtime()))
```

And don't forget to add the `import calendar`

Reply:
OK  your work very efficient.

Reply:
Using ``time.time()`` is much simpler.